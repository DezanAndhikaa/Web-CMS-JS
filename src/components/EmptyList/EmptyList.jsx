import React from 'react';
import { Empty } from '../../assets/imgs';
import './EmptyList.scss';

export default class EmptyList extends React.PureComponent {
    render() {
        return(
            <div className="paper-empty-list">
                <img className="empty-image" src={Empty} alt="" />
                <div className="caption-image">
                    <p className="header-caption">
                        {this.props.idEmpty === "Sales" ? "Sales Order List" :
                        this.props.idEmpty === "Rev" ? "Revision List" :
                        this.props.idEmpty === "Input" ? "Lifetime Component List" :
                        this.props.idEmpty === "Approve" ? "Approved List" :
                        this.props.idEmpty === "NA" ? "Not Approved List" :
                        this.props.idEmpty === "Delete" ? "Deleted List" :
                        this.props.idEmpty === "SAP" ? "SAP Issue List" : "Service Order List"}
                    </p>
                    <p className="caption1">You still don't have the 
                        {this.props.idEmpty === "Sales" ? " sales order list" :
                        this.props.idEmpty === "Rev" ? " revision" :
                        this.props.idEmpty === "Input" ? " lifetime component list" :
                        this.props.idEmpty === "Approve" ? " approved list" :
                        this.props.idEmpty === "NA" ? " not approved list" :
                        this.props.idEmpty === "Delete" ? " deleted list" :
                        this.props.idEmpty === "SAP" ? " SAP Issue" : " service order list"} record,
                    </p>
                    <p className="caption2">Let's keep it</p>
                </div>
            </div>
        )
    }
}