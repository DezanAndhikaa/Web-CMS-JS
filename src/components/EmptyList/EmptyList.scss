@import '../../themes/variables';
@import '../../themes/functions';

.paper-empty-list{
    margin-top: vw(35);
    border: 1.5px solid #E0E0E0;
    background-color: transparent;
    border-radius: 8px;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;
    .empty-image{
        width: vw(300);
        margin-top: vw(40);
    }
    .caption-image{
        margin: 0 0 vw(40) vw(50);
        .header-caption{
            margin-top: vw(20);
            font-size: vw(30) !important;
            line-height: vw(13) !important;
            font-family: 'Lato-Bold';
            color: #000000;
        }
        .caption1{
            font-size: vw(16) !important;
            line-height: vw(13) !important;
            font-family: 'Lato-Regular';
            color: #3C3B3B;
        }
        .caption2{
            font-size: vw(16) !important;
            line-height: vw(13) !important;
            font-family: 'Lato-Bold';
            color: #000000;
        }
    }
}