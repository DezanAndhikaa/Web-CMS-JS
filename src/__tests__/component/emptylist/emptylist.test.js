import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import EmptyList from '../../../components/EmptyList/EmptyList';

Enzyme.configure({adapter:new Adapter()});
it("render empty list correctly", () => {
    const wrapper = shallow (
        <EmptyList />
    )
    expect(wrapper).toMatchSnapshot();
})