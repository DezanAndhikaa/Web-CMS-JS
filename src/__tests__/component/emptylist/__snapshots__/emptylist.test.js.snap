exports[`render empty list correctly 1`] = `ShallowWrapper {}`;